src/http/h1/decode.rs

```diff
@@ -136,113 +126,177 @@ impl Decoder {
     }
 }
 
-fn eat<R: Read>(rdr: &mut R, bytes: &[u8]) -> io::Result<()> {
-    let mut buf = [0];
-    for &b in bytes.iter() {
-        match try!(rdr.read(&mut buf)) {
-            1 if buf[0] == b => (),
+macro_rules! byte (
+    ($rdr:ident) => ({
+        let mut buf = [0];
+        match try!($rdr.read(&mut buf)) {
+            1 => buf[0],
             _ => return Err(io::Error::new(io::ErrorKind::InvalidInput,
-                                          "Invalid characters found")),
+                                                "Invalid chunk size line")),
```

Comment:
Since this is saying that a read of size 1 is required, seems the error is more correctly thought of "unexpected eof during chunk size line". So it could now be represented as `io::ErrorKind::UnexpectedEof`.